server/routerlicious/packages/services-utils/src/asyncLocalStorage.ts

```diff
@@ -0,0 +1,25 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { AsyncLocalStorage } from "async_hooks";
+import * as uuid from "uuid";
+import { Request, Response, NextFunction } from "express";
+
+const asyncLocalStorage = new AsyncLocalStorage();
+
+export function getCorrelationId(): string | undefined {
+    const id = asyncLocalStorage.getStore();
+    return typeof(id) === "string" ? id : undefined;
```

Comment:
Technically we only store id as a string into the asyncLocalStorage instance. So when we get the stored value out, it should only be a string. My concern was when we call `asyncLocalStorage.getStore()`, it doesn't have any restrictions on the return type. So just added this type check to be cautious instead of casting it directly to string type. That's my thinking but I'm open to other opinions.

Reply:
Can we add the constraint like this?

const asyncLocalStorage: AsyncLocalStorage<{ "id": string }> = new AsyncLocalStorage();

Reply:
We can do `const asyncLocalStorage = new AsyncLocalStorage<T>();`, but the return type of `asyncLocalStorage.getStore()` is `T | undefined`. The purpose of this is when `getStore()` is called outside of `asyncLocalStorage.run()`, it won't get contexts and will return undefined. But we are going to do a type check anyway.

Reply:
Or we can do `const asyncLocalStorage = new AsyncLocalStorage<T>();` and just return the result of `getStore()` without checking type. Maybe that would be better.